backend/hitas/views/apartment.py

```diff
@@ -141,8 +142,74 @@ def validate(self, data):
         if start > end:
             raise ValidationError({"start": "'shares.start' must not be greater than 'shares.end'."})
 
+        apartment_filter = Q(building__real_estate__housing_company__uuid=F("_housing_company"))
+        if self.parent.instance is not None:
+            apartment_filter &= ~Q(uuid=self.parent.instance.uuid)
+
+        building_id: Optional[str] = None
+        building_input = self.parent.initial_data["building"]
+        if isinstance(building_input, dict):
+            building_id = building_input.get("id")
```

Comment:
Apartment will always have a `building_id` so no need to check if it exists

Reply:
These checks were added here after I ran into some issues with pre-existing tests: https://github.com/City-of-Helsinki/hitas/blob/master/backend/hitas/tests/apis/test_api_housing_company.py#L565

Could just remove the test cases if these aren't real checks that need to be made, but then giving `building_id` in an invalid form would crash instead of giving a useful error. What do you think?